### ISSUE TITLE
Lint/LiteralInInterpolation incorrect autocorrect for interpolated hash with quoted symbols

### ISSUE THREAD

**Issue by RedZapdos123** [rubocop/rubocop#15291]
## Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

## Issue type

Incorrect autocorrect

## Cop name

`Lint/LiteralInInterpolation`

## Steps to reproduce the problem

1. Create a file `example.rb` with:

```ruby
puts "#{ { :'foo-bar' => 1 } }"
```

2. Run:

```sh
bundle exec ruby exe/rubocop example.rb --only Lint/LiteralInInterpolation -A
```

3. RuboCop autocorrects it to:

```ruby
puts "{:foo-bar=>1}"
```

4. Compare behavior before and after:

Before:

```text
{:"foo-bar"=>1}
```

After:

```text
{:foo-bar=>1}
```

## Expected behavior

Autocorrect should preserve the stringified hash representation.

For this case it should keep the quoted symbol form inside the interpolated hash:

```ruby
puts "{:\"foo-bar\"=>1}"
```

or equivalently avoid autocorrecting if the cop cannot serialize the symbol safely.

## Actual behavior

Autocorrect changes the behavior by serializing `:'foo-bar'` as `:foo-bar` inside the hash string.

## RuboCop configuration

No special configuration.

## RuboCop version

```shell
$ bundle exec ruby exe/rubocop -V
1.87.0 (using Parser 3.3.11.1, rubocop-ast 1.49.1, analyzing as Ruby 2.7, running on ruby 3.2.3) +Rubydex [x86_64-linux-gnu]
  - rubocop-performance 1.26.1
  - rubocop-rake 0.7.1
  - rubocop-rspec 3.10.2
```

**Comment by RedZapdos123** [comment]
Opened a PR to resolve this issue.
